tsMuxer/CMakeLists.txt

```diff
@@ -116,6 +116,76 @@ if(TSMUXER_STATIC_BUILD)
   endif()
 endif()
 
+# we can try compiling this small C++ program to test std::filesystem is working
+string(CONFIGURE [[
+  #include <filesystem>
+  int main( int argc, char ** argv ) {
+    std::filesystem::path p( argv[ 0 ] );
+    return p.string().length();
+  }
+]] STD_FS_CODE @ONLY)
+
+string(CONFIGURE [[
+  #include <experimental/filesystem>
+  int main( int argc, char ** argv ) {
+    std::experimental::filesystem::path p( argv[ 0 ] );
+    return p.string().length();
+  }
+]] EXP_FS_CODE @ONLY)
+
+string(CONFIGURE [[
+  #include <ghc/filesystem.hpp>
+  int main( int argc, char ** argv ) {
+    ghc::filesystem::path p( argv[ 0 ] );
+    return p.string().length();
+  }
+]] GHC_FS_CODE @ONLY)
+
+# required for MacOS, osxcross and LLVM Clang
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS "-std=c++17")
```

Comment:
This doesn't feel right. The main CMake file already has
```
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
set(CMAKE_CXX_EXTENSIONS FALSE)
```
Which should result in `-std=c++17` being there already. I'll investigate and report back - I suspect `std=c++17` isn't the cause of trouble here.

Reply:
Okay. For some reason `-std=c++17` doesn't reach the complier command line when compiling on Catalina. According to [the documentation](https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html), the variable `CMAKE_REQUIRED_FLAGS` can be set to force including a flag, which feels less messy than modifying `CMAKE_CXX_FLAGS` directly. Worked for me.